src/frontend/src/icp-eth/components/tokens/ManageTokensModal.svelte

```diff
@@ -50,6 +57,20 @@
 		]);
 	};
 
+	const saveErc20Token = async () => {
+		await saveErc20Contract({
+			contractAddress: erc20ContractAddress,
+			metadata: erc20Metadata,
+			chainId: $selectedChainId,
```

Comment:
I think we have an issue here on "Chain Fusion" because`$selectedChainId` and `$selectedEthereumNetwork` are either undefined or ethereum (did not checked the code). I guess they should be related to the network the user has selected when they added the contract manually no?

Reply:
yes! it was an overview on my side, correcting it

Reply:
Good idea